src/Polly.Shared/NoOp/NoOpEngine.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Polly.Shared.NoOp
+{
+    internal static partial class NoOpEngine
+    {
+        internal static TResult Implementation<TResult>(Func<CancellationToken, TResult> action,
+            CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            DelegateResult<TResult> delegateOutcome = 
+                new DelegateResult<TResult>(action(cancellationToken));
```

Comment:
Could remove the `DelegateResult<TResult>` allocation.  The `delegateOutcome.Result` below only returns the `action(cancellationToken)` that has just been passed in to the ctor.  

(similar where same pattern on async side)